### ISSUE TITLE
`brew upgrade` doesn't install the new cask for emacs-27.1-mac-8.1 containing multi-tty support

### ISSUE THREAD

**Issue by jlokier** [railwaycat/homebrew-emacsmacport#227]
Woohoo!  Multi-tty support while running on GUI, lovely!  Even my 24-bit colour themes appear in the terminal looking virtually identical to a GUI window, except they are in a much more usable place for editing command-line things like Git commit logs and other `$EDITOR` usages :-)

However...

As far as I can tell, for `brew install --cask emacs-mac` users, multi-tty support was missing in the cask shipped recently on 2020-11-15 for `emacs-27.1-mac-8.1`, but it landed correctly in the more recent cask update on 2020-11-21 _also_ for `emacs-27.1-mac-8.1`.

So 2020-11-21 was a nice update.  But there was a little problem: Homebrew didn't upgrade to it.

`brew update` highlighted `emacs-mac` as having a new cask.  But `brew upgrade` did not upgrade it (it did upgrade other things including other casks).  Being more explicit, `brew upgrade --cask emacs-mac` did not upgrade either. I double checked just in case it was up to date; it wasn't, the multi-tty support wasn't in the installed binary or `server.el`.

Turns out to upgrade to the latest cask binary I had to uninstall and reinstall:

```
Jamies-MBP:~ jamie$ brew info --cask emacs-mac

emacs-mac: emacs-27.1-mac-8.1
https://bitbucket.org/mituharu/emacs-mac.git
/Users/jamie/brew/Caskroom/emacs-mac/emacs-27.1-mac-8.1 (119B)
From: https://github.com/railwaycat/homebrew-emacsmacport/blob/HEAD/Casks/emacs-mac.rb
==> Name
Emacs-mac
==> Description
None
==> Artifacts
Emacs.app (App)
/Applications/Emacs.app/Contents/MacOS/Emacs.sh -> emacs (Binary)
/Applications/Emacs.app/Contents/MacOS/bin/ebrowse (Binary)
/Applications/Emacs.app/Contents/MacOS/bin/emacsclient (Binary)
/Applications/Emacs.app/Contents/MacOS/bin/etags (Binary)
/Applications/Emacs.app/Contents/Resources/include/emacs-module.h -> /Users/jamie/brew/include/emacs-module.h (Binary)

Jamies-MBP:~ jamie$ brew remove --cask emacs-mac

==> Uninstalling Cask emacs-mac
==> Backing App 'Emacs.app' up to '/Users/jamie/brew/Caskroom/emacs-mac/emacs-27.1-mac-8.1/Emacs.app'.
==> Removing App '/Applications/Emacs.app'.
==> Unlinking Binary '/Users/jamie/brew/bin/emacs'.
==> Unlinking Binary '/Users/jamie/brew/bin/ebrowse'.
==> Unlinking Binary '/Users/jamie/brew/bin/emacsclient'.
==> Unlinking Binary '/Users/jamie/brew/bin/etags'.
==> Unlinking Binary '/Users/jamie/brew/include/emacs-module.h'.
==> Purging files for version emacs-27.1-mac-8.1 of Cask emacs-mac

Jamies-MBP:~ jamie$ brew install --cask emacs-mac

==> Downloading https://github.com/railwaycat/homebrew-emacsmacport/releases/download/emacs-27.1-mac-8.1/e
Already downloaded: /Users/jamie/Library/Caches/Homebrew/downloads/75ef8e0137518ad6ec3a937e77dd02d15c0905d48c86f752bb4ded2249372649--emacs-27.1-mac-8.1-10.14.6.zip
==> Installing Cask emacs-mac
==> Moving App 'Emacs.app' to '/Applications/Emacs.app'.
==> Linking Binary 'Emacs.sh' to '/Users/jamie/brew/bin/emacs'.
==> Linking Binary 'ebrowse' to '/Users/jamie/brew/bin/ebrowse'.
==> Linking Binary 'emacsclient' to '/Users/jamie/brew/bin/emacsclient'.
==> Linking Binary 'etags' to '/Users/jamie/brew/bin/etags'.
==> Linking Binary 'emacs-module.h' to '/Users/jamie/brew/include/emacs-module.h'.
🍺  emacs-mac was successfully installed!

Jamies-MBP:~ jamie$ brew info --cask emacs-mac

emacs-mac: emacs-27.1-mac-8.1
https://bitbucket.org/mituharu/emacs-mac.git
/Users/jamie/brew/Caskroom/emacs-mac/emacs-27.1-mac-8.1 (119B)
From: https://github.com/railwaycat/homebrew-emacsmacport/blob/HEAD/Casks/emacs-mac.rb
==> Name
Emacs-mac
==> Description
None
==> Artifacts
Emacs.app (App)
/Applications/Emacs.app/Contents/MacOS/Emacs.sh -> emacs (Binary)
/Applications/Emacs.app/Contents/MacOS/bin/ebrowse (Binary)
/Applications/Emacs.app/Contents/MacOS/bin/emacsclient (Binary)
/Applications/Emacs.app/Contents/MacOS/bin/etags (Binary)
/Applications/Emacs.app/Contents/Resources/include/emacs-module.h -> /Users/jamie/brew/include/emacs-module.h (Binary)
```

That worked: The binaries were updated.

So this issue is to report that the new cask isn't picked up by users doing `brew upgrade`. Maybe Homebrew looks at the version string and decides there's no need to upgrade when the version string `emacs-27.1-mac-8.1` hasn't changed? Assuming yes, fix would be to bump it to 8.2.

**Comment by rprimus** [comment]
Fri Nov 27 14:14:53 GMT 2020

> fix would be to bump it to 8.2

When the version hasn't changed, any changes are noted by appending '_<n>' (where n is the change number) eg:
`emacs-27.1-mac-8.1_1`.

**Comment by railwaycat** [comment]
Hi @jlokier, thanks to let me know! It's my fault I didn't noticed that Travis CI will not update the build results of draft tag if the tag already exists, then I messed up two draft results when running script to sign and release them. 

I apology for the inconvenience and I will more careful next time. 

@rprimus is correct on the version number scheme.
